hazelcast/src/main/java/com/hazelcast/sql/impl/expression/math/ExpressionMath.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.math;
+
+import com.hazelcast.sql.SqlErrorCode;
+import com.hazelcast.sql.impl.QueryException;
+
+import java.math.MathContext;
+import java.math.RoundingMode;
+
+import static com.hazelcast.sql.impl.type.QueryDataType.MAX_DECIMAL_PRECISION;
+
+/**
+ * Utility methods for math functions.
+ */
+public final class ExpressionMath {
+
+    /**
+     * Math context used by expressions while doing math on BigDecimal values.
+     */
+    public static final MathContext DECIMAL_MATH_CONTEXT = new MathContext(MAX_DECIMAL_PRECISION, RoundingMode.HALF_DOWN);
```

Comment:
Is there a specific reason why we picked `HALF_DOWN` here, and not `HALF_UP` or `HALF_EVEN`? Can we have JavaDoc explaining this decision?

Reply:
I believe I just copied it blindly from the pre-existing prototype implementation assuming it's correct. Looks like the standard doesn't specify any defaults for rounding/truncation modes, so it's implementation defined. I think `HALF_UP` might be a better candidate since that's the mode most users known from school. WDYT?

Reply:
Agree, HALF_UP appears to be a good fit.

Reply:
fixed